api/fbs/v1/write.fbs

```diff
@@ -0,0 +1,66 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+include "database.fbs";
+
+namespace banyandb.v1;
+
+table String {
+    value: string;
+}
+
+table Int {
+    value: int64;
+}
+
+union ValueType {
```

Comment:
Is it possible/necessary to distinguish single-value and multi-value data types? For example: `String`, `StringArray`? 

For now, it seems every value is a vector which may not be efficient in the storage layer.

Reply:
Your path might reduce the data size theoretically, but I don't know how much it will help get a more compacted data format.

Since banyandb's core value is to find a more effective way to store trace data, this approach makes sense to me. Let's introduce `StringArray` and `IntArray` here